src/openforms/submissions/logic/actions.py

```diff
@@ -245,13 +249,23 @@ def eval(
             for item in self.input_mapping
         }
 
-        # Perform DMN call
-        dmn_outputs = evaluate_dmn(
-            definition_id=self.decision_definition_id,
-            version=self.decision_definition_version,
-            input_values=dmn_inputs,
-            plugin_id=self.plugin_id,
+        def _evaluate_dmn():
+            return evaluate_dmn(
+                definition_id=self.decision_definition_id,
+                version=self.decision_definition_version,
+                input_values=dmn_inputs,
+                plugin_id=self.plugin_id,
+            )
+
+        # Perform DMN call or retrieve result from cache
+        cache_key = hash(
+            str(submission.uuid)
+            + self.decision_definition_id
+            + self.decision_definition_version
+            + self.plugin_id
+            + json.dumps(serialize_variables(dmn_inputs), sort_keys=True)
         )
```

Comment:
instead of `serialize_variables`, you can probably leverage `from django.core.serializers.json import DjangoJSONEncoder` :sweat_smile: 

I'd then also add some type hints that incorporate `openforms.typing.DjangoJSONEncodable` so we have type-checking taking care of which variable types are serializable (or at least have the intent documented).